### DISCUSSION TITLE
Fail to ingest Palo Alto Logs in SO Version 2.3.130

### DISCUSSION THREAD

**Discussion by sikelos** [Security-Onion-Solutions/securityonion#8392] [Unsupported Versions] [answered]
SO version 2.3.130
Standalone
I am following the procedure described in 
https://docs.securityonion.net/en/2.3/filebeat.html?highlight=netflow#walkthrough-netflow-logs
and in the SO video
https://www.youtube.com/watch?v=ew5gtVjAs7g
to ingest Palo Alto logs in Filebeat.
The process stops on the 2nd step when I run the command 
salt-call state.apply filebeat

I receive the error **"port is already allocated"**

![image](https://user-images.githubusercontent.com/25420601/181578677-9d61e9af-1349-41f5-8a70-bd68f9496801.png)

I **tried several different ports** ending to the same error (Attached)

The Filebeat container then goes from "Pending" to "Missing" in **so-status**

Running the **so-filebeat-start** complains for missing container

The **/opt/so/log/master** shows this

![image](https://user-images.githubusercontent.com/25420601/181579032-d04bf5b0-bd00-431c-9053-2c6a40949866.png)

A **salt-call state.highstate**  gives the same 2 errors:

![image](https://user-images.githubusercontent.com/25420601/181581687-d4b4d60c-ae44-4625-9650-15d6d8399265.png)

I just ran now the same exact steps in a fresh install ending again to same errors.

Any suggestions please as I am on this issue for the last 3 days at least trying all different solutions found on the Security Onion discussions.

**Comment by reyesj2** [comment]
What is the output of:
```
sudo docker ps | grep so-filebeat
```
Can you also share a sanitized minion pillar file and filebeat init.sls

**Reply by sikelos** [reply]
No output possibly because the so-status shows that now the so-filebeat container is missing

![image](https://user-images.githubusercontent.com/25420601/181605924-72229eea-24ec-40e9-a43b-de7c86e3e44d.png)

**Reply by sikelos** [reply]
I 've ran through the same cycle as I wrote before:  after adding the filebeat settings I ve got the initial errors and then the filebeat container goes from PENDING to MISSING
Then I have to return with a snapshot I got before adding the settings, to the previous sane state, in order to test again

**Reply by reyesj2** [reply]
Maybe post your filebeat configs, for help spotting syntax errors. When applying the configuration changes it fails. When you roll back changes it runs?

**Reply by sikelos** [reply]
When I roll back everything is green is so-status yes

This is from /opt/so/saltstack/local/pillar/minios/siem_standalone.sls
(I tried several ports...same error)

![image](https://user-images.githubusercontent.com/25420601/181620359-43f4c16d-1b3d-4818-ac6b-3ca3e20b7032.png)


This is from /opt/so/saltstack/local/salt/filebeat/init.sls (Copied from the default directory as per the steps from SO page)

![image](https://user-images.githubusercontent.com/25420601/181620388-2945c440-25e5-49f3-8f07-0a4806ff6be9.png)

**Comment by THExCAPNxRK** [comment, marked as answer]
I ran into this issue a few updates ago. The fix for me was remove the port binding from the init.sls file. Enabling the module with automatically bind the port.

You can verify the port is binded by running `docker ps | grep 9002` once you restarted filebeat.

Hope this helps.

**Reply by sikelos** [reply]
Thanks for the info. I will try it but what do you mean with "Enabling the module with automatically bind the port."?
That means I do only step 1 from the 
https://docs.securityonion.net/en/2.3/filebeat.html?highlight=netflow#walkthrough-netflow-logs
Do you remember the steps you followed?

**Reply by THExCAPNxRK** [reply]
Sorry for the confusion, the only step I DIDNT do was step 2 where you edit the docker config file, so following that walkthrough the steps I took were the following.

1. Enabled the third part module
2. Update the firewall config
3. Build logstash pipeline

**Reply by sikelos** [reply]
Thanks. I will try it and update if this works

**Reply by sikelos** [reply]
I did step one (Assigned port 9002 for filebeat module and firewall settings (Step 3) , in /opt/so/saltstack/local/pillar/minios/siem_standalone.sls) . It looks indeed is updating the port bindings for the filebeat

Then when applying with "salt-call state.apply firewall"  comes with the error "data failed to compile" 

![image](https://user-images.githubusercontent.com/25420601/182146605-17fc36ff-956c-451a-9781-5690970097bd.png)

By commenting out the firewall additions in sls file , the state.apply command runs without errors. Do you see any syntax errors here?

![image](https://user-images.githubusercontent.com/25420601/182151885-c395931e-646c-4cc7-a1f7-883b853ebd1e.png)

**Reply by THExCAPNxRK** [reply]
Here is a screenshot of what I have, Disclaimer mine is setup to go to a sensor so not sure if that makes a difference.
![image](https://user-images.githubusercontent.com/83655473/182155897-c469e390-194c-427e-9a7f-042708bb80fc.png)

If that doesn't help the salt loga should have some clues
`/opt/so/log/salt/`

**Reply by sikelos** [reply]
Ok it looks like that the steps you suggested worked 
I have finally logs in Kibana for the module "panw" (Palo Alto Logs) I created earlier 

![image](https://user-images.githubusercontent.com/25420601/182200620-e1232210-f01a-4d6e-b15e-971e0f3ebc42.png)


Thanks a lot @THExCAPNxRK  for the CRUCIAL suggestion!!!

If I will get some extra info on this issue, I will add it here

**Comment by TotieBash** [comment]
is it in the local registry?  docker images | grep filebeat

if it is in there try:  so-filebeat-restart

I had similar issue with missing so-redis container but when I do the so-redis-restart it has checking mechanism there to realize so-redis is missing so it then pulls it from local docker registry.  Hopefully it has the same mechanism.
